client/client-lib/src/mint/mod.rs

```diff
@@ -241,39 +268,13 @@ impl<'c> MintClient<'c> {
     }
 }
 
-impl NoteIssuanceRequests {
-    /// Generate a new `IssuanceRequest` and the associates [`SignRequest`]
-    pub fn new<K, C>(
-        amount: Amount,
-        amount_tiers: &Tiered<K>,
-        ctx: &Secp256k1<C>,
-        rng: &mut (impl RngCore + CryptoRng),
-    ) -> (NoteIssuanceRequests, SignRequest)
-    where
-        C: Signing,
-    {
-        let (requests, blinded_nonces): (TieredMulti<_>, TieredMulti<_>) =
-            TieredMulti::represent_amount(amount, amount_tiers)
-                .into_iter()
-                .map(|(amt, ())| {
-                    let (request, blind_msg) = NoteIssuanceRequest::new(ctx, rng);
-                    ((amt, request), (amt, blind_msg))
-                })
-                .unzip();
-
-        debug!(
-            %amount,
-            coins = %requests.item_count(),
-            tiers = ?requests.tiers().collect::<Vec<_>>(),
-            "Generated issuance request"
-        );
-
-        let sig_req = SignRequest(blinded_nonces);
-        let issuance_req = NoteIssuanceRequests { coins: requests };
-
-        (issuance_req, sig_req)
+impl Extend<(Amount, NoteIssuanceRequest)> for NoteIssuanceRequests {
```

Comment:
TIL: `Extend`

Reply:
It's needed for `unzip`, confuses me every time that `FromIterator` doesn't work for that.